middlewares/hacker.middleware.js

```diff
@@ -215,6 +216,27 @@ async function updateHacker(req, res, next) {
     }
 }
 
+/**
+ * Checks that there are no other hackers with the same account id as the one passed into req.body.accountId
+ * @param {{body:{accountId: ObjectId}}} req 
+ * @param {*} res 
+ * @param {*} next
+ */
+async function checkDuplicateAccountLinks(req, res, next) {
+    const hackers = await Services.Hacker.findByAccountId(req.body.accountId);
+    if (!hackers || hackers.length === 0) {
+        next();
+    } else {
+        next({
+            status: 404,
```

Comment:
Should not be 404, should be something along the lines of 422 (validation failed), or something.

Reply:
409 - Conflict

Reply:
Changed to 409